src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -457,3 +461,31 @@ export function disabledOptionToKeyedTuple(
     return [valueToKey(disabledOption), undefined];
   }
 }
+
+type AtLeastOne<T, U = { [K in keyof T]: Pick<T, K> }> = Partial<T> & U[keyof U];
+
+type InferrableOptionValue = AtLeastOne<{
+  id: any;
+  value: any;
```

Comment:
Thought: granted, this is not a public API, but would prefer using `Value` instead of `any`.